tlsdialer.go

```diff
@@ -33,8 +34,14 @@ type Dialer struct {
 	Timeout        time.Duration
 	Network        string
 	SendServerName bool
+
 	// Force validation of a specific name other than the SNI name or dialed hostname
-	ForceValidateName  string
+	ForceValidateName string
+
+	// Must be provided if ClientHelloID is set to tls.HelloCustom. This field is ignored if
+	// ClientHelloID is not set to tls.HelloCustom.
+	ClientHelloSpec *tls.ClientHelloSpec
```

Comment:
To be super safe, maybe also check if ClientHelloSpec is specified but ClientHelloID is not HelloCustom?

Reply:
Sure, I can do that